shared/actions/chat.js

```diff
@@ -238,6 +236,20 @@ function * _incomingMessage (action: IncomingMessage): SagaGenerator<any, any> {
         const conversationIDKey = conversationIDToKey(incomingMessage.convID)
         const message = _unboxedToMessage(messageUnboxed, 0, yourName, conversationIDKey)
 
+        // Is this message for the currently selected conversation? If so, mark
+        // it as read ASAP to avoid badging it -- we don't need to badge, the
+        // user's looking at it already.
+        const selectedSelector = (state: TypedState) => state.chat.get('selectedConversation')
+        const selectedConversationIDKey = yield select(selectedSelector)
+        if (message && message.messageID && conversationIDKey === selectedConversationIDKey) {
```

Comment:
slightly more complicated than this. we only mark it as read if the app has focus. @gabriel added this to the store last week